hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaRolloutManagement.java

```diff
@@ -760,9 +761,13 @@ private boolean isRolloutComplete(final JpaRollout rollout) {
     }
 
     private boolean isRolloutGroupComplete(final JpaRollout rollout, final JpaRolloutGroup rolloutGroup) {
-        final Long actionsLeftForRollout = actionRepository
-                .countByRolloutAndRolloutGroupAndStatusNotAndStatusNotAndStatusNot(rollout, rolloutGroup,
-                        Action.Status.ERROR, Action.Status.FINISHED, Action.Status.CANCELED);
+        List<Status> terminalStatuses = new LinkedList<>(Arrays.asList(Status.ERROR, Status.FINISHED, Status.CANCELED));
```

Comment:
To avoid creating and modifying the list on every call two static unmodifiable lists could be used.